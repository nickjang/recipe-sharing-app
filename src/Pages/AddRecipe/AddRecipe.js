import React, { Component } from 'react';
import RecipeForm from '../../Components/RecipeForm/RecipeForm';
import './AddRecipe.css';

class AddRecipe extends Component {
  handleAddSuccess = (recipeId) => {
    this.props.history.push(`/recipes/${recipeId}`);
  }

  render() {
    return (
      <RecipeForm
        onSuccess={this.handleAddSuccess}
        title='Add Recipe'
        form='add-recipe-form'
        submitButtonText='Add recipe' />
    );
  }
}

AddRecipe.defaultProps = {
  location: {},
  history: {
    push: () => { }
  }
}

export default AddRecipe;