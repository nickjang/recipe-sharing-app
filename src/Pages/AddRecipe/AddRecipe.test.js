import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddRecipe from './AddRecipe';

describe(`AddRecipe component`, () => {
  it('renders AddRecipe by default', () => {
    const wrapper = shallow(<AddRecipe />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})