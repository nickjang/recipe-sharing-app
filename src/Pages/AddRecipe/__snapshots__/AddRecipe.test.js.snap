// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe component renders AddRecipe by default 1`] = `
<Fragment>
  <withRouter(RecipeForm)
    fetchRecipe={false}
    form="add-recipe-form"
    onSuccess={[Function]}
    submitButtonText="Add recipe"
    title="Add Recipe"
  />
</Fragment>
`;
